src/app/components/Icons/Icons.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import DropIcon from '../assets/Drop';
+import LeafIcon from '../assets/Leaf';
+
+export type IconProps = {
+  iconType: 'dropBig' | 'dropSmall' | 'leafBig' | 'leafSmall';
+};
+
+const Icons = ({ iconType }: IconProps): JSX.Element => {
+  const iconSizeMap = {
+    dropBig: <DropIcon width="1.1813rem" height="2.438rem" />,
+    dropSmall: <DropIcon width="1rem" height="0.938rem" />,
+    leafBig: <LeafIcon width="2.492rem" height="1.813rem" />,
+    leafSmall: <LeafIcon width="1.115rem" height="0.773rem" />,
+  };
+
+  return <div>{iconSizeMap[iconType]}</div>;
```

Comment:
Not sure if this could happen because typescript is telling you something is missing, but whould be is a `default` case useful?

```suggestion
  const iconSizeMap = {
    dropBig: <DropIcon width="1.1813rem" height="2.438rem" />,
    dropSmall: <DropIcon width="1rem" height="0.938rem" />,
    leafBig: <LeafIcon width="2.492rem" height="1.813rem" />,
    leafSmall: <LeafIcon width="1.115rem" height="0.773rem" />,
    default: <DefaultIcon width="1.115rem" height="0.773rem" />,
  };

  return <div>{iconSizeMap[iconType] || iconSizeMap["default"]}</div>;
```

I'm not so knowledgeable in typescript. Maybe @lmachens knows more about that?

Reply:
A default case should not be required because `iconType`  is defined as `'dropBig' | 'dropSmall' | 'leafBig' | 'leafSmall'`.
Thus, you would get an error if you set a different type.

I would define type of `iconSizeMap`:

```tsx
const iconSizeMap: {
  [iconType in IconProps["iconType"]]: JSX.Element
} = {
    dropBig: <DropIcon width="1.1813rem" height="2.438rem" />,
    dropSmall: <DropIcon width="1rem" height="0.938rem" />,
    leafBig: <LeafIcon width="2.492rem" height="1.813rem" />,
    leafSmall: <LeafIcon width="1.115rem" height="0.773rem" />,
  };
```

To optimize this, I would create `IconType` type and move `iconSizeMap` to prevent recreating this object on each render.

```tsx
type IconType = 'dropBig' | 'dropSmall' | 'leafBig' | 'leafSmall';

type IconProps = {
  iconType: IconType;
};

const iconSizeMap: {
  [iconType in IconType]: JSX.Element
} = {
    dropBig: <DropIcon width="1.1813rem" height="2.438rem" />,
    dropSmall: <DropIcon width="1rem" height="0.938rem" />,
    leafBig: <LeafIcon width="2.492rem" height="1.813rem" />,
    leafSmall: <LeafIcon width="1.115rem" height="0.773rem" />,
  };

const Icons = ({ iconType }: IconProps): JSX.Element => {
   return <div>{iconSizeMap[iconType]}</div>;
}
```

Reply:
Thank you both much for your review